### ISSUE TITLE
How to use the gradle plugin to use snapshot ?

### ISSUE THREAD

**Issue by Lakedaemon** [Prototik/HoloEverywhere#759]
I managed to set it up to use 2.1.0 but I would like to be able to use 2.1.1-snapshot (in part because it uses support 19 instead of support 18.0.4)

what should I write in my gradle files to make the holoeverywhere 0.2.+ plugin use the latest snapshot ? This isn't explained in the wiki

I tried a few things (library{Version '2.1.1-snapshot}} in build.gradle, looking at the code of the plugin in mavenCentral and adding include 'holoeverywhere-snapshot' in settings.gradle, ... but nothing worked for me

**Comment by Prototik** [comment]
First you should enable snapshot repository:

```
holoeverywhere {
  repository {
    snapshot 'yes'
  }
  library {
    version '2.1.1-SNAPSHOT'
  }
}
```

Please describe here if it works.

**Comment by Lakedaemon** [comment]
On 01/18/2014 01:23 PM, Sergey Shatunov wrote:

> |holoeverywhere {
>    repository {
>      snapshot 'yes'
>    }
>    library {
>      version '2.1.1-SNAPSHOT'
>    }
> }|
> I have had mitigated success :

I tried it and I am sure that it worked at least once because i saw a

prepareOrgHoloeverywhereAddonPreferences211SNAPSHOTLibrary

entry in my gradle console.

But as I still saw the Gradle-support-v4.18.04.jar and 
holoeverywhere210.jar in my IDEA libraries (and not 19 and 2.1.1snapshot),
I cleaned up my gradle and idea configurations a bit and then tried 
again to pull 2.1.1-snapshot in my libraries without success

Now, I can only get

prepareOrgHoloeverywhereAddonPreferences210Library

This is really strange.

Could there be something wrong with my gradle.build file ?

buildscript {
     repositories {
         mavenCentral()
         maven {url 
'http://oss.sonatype.org/content/repositories/snapshots'}
        }

```
 dependencies {
     classpath ext.android_tools_gradle
     classpath ext.kotlin_plugin
     classpath 'org.holoeverywhere:plugin:0.2.+'
 }
```

}

apply plugin: 'holoeverywhere'
apply plugin: 'android-library'
apply plugin: 'kotlin-android'

repositories {
     mavenCentral()
     maven {
         url 'http://oss.sonatype.org/content/repositories/snapshots'
     }
}

dependencies {
     compile files('/home/ubuntu/idea/lib/annotations.jar')
     compile project(':common')
     compile project(':lucene')
     compile files('/home/ubuntu/git/tagsoup-1.2.1.jar') { 
ext.transitive = false }
     compile project(':compress') { ext.transitive = false }
     compile project(':googleDownloading') { ext.transitive = false }
     compile project(':swipeList') { ext.transitive = false }
     compile "org.jetbrains.kotlin:kotlin-stdlib:$ext.kotlin_version"
}

holoeverywhere {
     repository {
         snapshot 'yes'
     }
     addons {
         create 'preferences'
         //  create 'slider'
     }
     library {
         version '2.1.1-SNAPSHOT'
     }
}

android {
     compileSdkVersion 19
     buildToolsVersion "19.0.1"

```
 defaultConfig {
     minSdkVersion 9
     targetSdkVersion 19
     ndk {
         moduleName "aMemory"
         cFlags "-DANDROID_NDK -DDISABLE_IMPORTGL"
         ldLibs "log"
     }
     buildConfigField "boolean", "HAS_LOOKUP", "true"
 }

 sourceSets {
     main {
         manifest.srcFile 'src/main/AndroidManifest.xml'

         java {
             srcDirs += 'src/main/kotlin'
             exclude 'build/kotlin-classes/**'
         }
         res.srcDirs = ['src/main/res']
         assets.srcDirs = ['src/main/assets']
     }
 }

 lintOptions {
     abortOnError false
     // disable 'MissingTranslation'
 }
```

}

**Comment by Prototik** [comment]
mmm, yeah, you should also modify support version:

```
holoeverywhere {
  ...
  supportV4 {
    version = '19.0.3'
  }
}
```

**Comment by Lakedaemon** [comment]
I'll do it

But...isn't there a problem with this part in the plugin ?

if (!p.repositories.any { it.name == REPO_NAME }) {
                     p.repositories.maven {
                         name REPO_NAME
                         url holoeverywhere.repository.url
                     }
                 }
                 if (snapshot && !p.repositories.any { it.name == 
SNAPSHOT_REPO_NAME }) {
                     p.repositories.maven {
                         name SNAPSHOT_REPO_NAME
                         url holoeverywhere.repository.snapshotUrl
                     }
                 }

Do the boolean conditions work as intended ?
I mean : if I only apply the holoeverywhere plugin (without setting an 
holoeverywhere repository, or by setting an holoeverywhere-snapshot 
repository)
the first condition will be true and I'll get the 2.1.0 library, right ?

**Comment by Prototik** [comment]
In your buildscript above both repositories will attached to project.
But first repo doesn't have any snapshot artifacts, then if you force specify snapshot version of library will be using version from second repo.
btw gradle throw an exception if any unresolved dependency found, I sure that correct version of libraries using.
You can invoke `gradlew androidDependencies dependencies` and post output?

**Comment by Lakedaemon** [comment]
On 01/18/2014 04:29 PM, Sergey Shatunov wrote:

> In your buildscript above both repositories will attached to project.
> But first repo doesn't have any snapshot artifacts, then if you force 
> specify snapshot version of library will be using version from second 
> repo.
> btw gradle throw an exception if any unresolved dependency found, I 
> sure that correct version of libraries using.
> You can invoke |gradlew androidDependencies| and post output?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/Prototik/HoloEverywhere/issues/759#issuecomment-32684406.
> 
> Here is my output :
> ubuntu@ubuntu-W150HRM:~/git/android$ ./gradlew androidDependencies
> The TaskContainer.add() method has been deprecated and is scheduled to 
> be removed in Gradle 2.0. Please use the create() method instead.
> :core:androidDependencies
> debug
> +--- LOCAL: annotations.jar
> +--- LOCAL: tagsoup-1.2.1.jar
> +--- android:googleDownloading:unspecified
> |    --- android:googleLicensing:unspecified
> +--- android:swipeList:unspecified
> |    --- LOCAL: nineoldandroids-2.4.0.jar
> +--- org.holoeverywhere:library:2.1.0
> --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
+--- LOCAL: annotations.jar
+--- LOCAL: tagsoup-1.2.1.jar
+--- debug
|    +--- LOCAL: tagsoup-1.2.1.jar
|    +--- LOCAL: annotations.jar
|    +--- android:googleDownloading:unspecified
|    |    --- android:googleLicensing:unspecified
|    +--- android:swipeList:unspecified
|    |    --- LOCAL: nineoldandroids-2.4.0.jar
|    +--- org.holoeverywhere:library:2.1.0
|    --- org.holoeverywhere:addon-preferences:2.1.0
+--- android:googleDownloading:unspecified
|    --- android:googleLicensing:unspecified
+--- android:swipeList:unspecified
|    --- LOCAL: nineoldandroids-2.4.0.jar
+--- org.holoeverywhere:library:2.1.0
--- org.holoeverywhere:addon-preferences:2.1.0

release
+--- LOCAL: annotations.jar
+--- LOCAL: tagsoup-1.2.1.jar
+--- android:googleDownloading:unspecified
|    --- android:googleLicensing:unspecified
+--- android:swipeList:unspecified
|    --- LOCAL: nineoldandroids-2.4.0.jar
+--- org.holoeverywhere:library:2.1.0
--- org.holoeverywhere:addon-preferences:2.1.0
:dLight:androidDependencies
debug
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:dPro:androidDependencies
debug
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:googleDownloading:androidDependencies
debug
--- android:googleLicensing:unspecified

debugTest
+--- debug
|    --- android:googleLicensing:unspecified
--- android:googleLicensing:unspecified

release
--- android:googleLicensing:unspecified
:googleLicensing:androidDependencies
debug
No dependencies

debugTest
--- debug

release
No dependencies
:rPro:androidDependencies
debug
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:unspecified
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:swipeList:androidDependencies
debug
--- LOCAL: nineoldandroids-2.4.0.jar

debugTest
+--- LOCAL: nineoldandroids-2.4.0.jar
--- debug
      --- LOCAL: nineoldandroids-2.4.0.jar

release
--- LOCAL: nineoldandroids-2.4.0.jar

BUILD SUCCESSFUL

**Comment by Lakedaemon** [comment]
I can also get  (there are a few 2.1.1-Snapshot in there....but not at 
the right places i think)

ubuntu@ubuntu-W150HRM:~/git/android$ ./gradlew androidDependencies
The TaskContainer.add() method has been deprecated and is scheduled to 
be removed in Gradle 2.0. Please use the create() method instead.
:core:androidDependencies
debug
+--- LOCAL: annotations.jar
+--- LOCAL: tagsoup-1.2.1.jar
+--- android:googleDownloading:unspecified
|    --- android:googleLicensing:unspecified
+--- android:swipeList:unspecified
|    --- LOCAL: nineoldandroids-2.4.0.jar
+--- org.holoeverywhere:library:2.1.0
--- org.holoeverywhere:addon-preferences:2.1.0

debugTest
+--- LOCAL: annotations.jar
+--- LOCAL: tagsoup-1.2.1.jar
+--- debug
|    +--- android:googleDownloading:unspecified
|    |    --- android:googleLicensing:unspecified
|    +--- android:swipeList:unspecified
|    |    --- LOCAL: nineoldandroids-2.4.0.jar
|    +--- org.holoeverywhere:library:2.1.0
|    --- org.holoeverywhere:addon-preferences:2.1.0
+--- android:googleDownloading:unspecified
|    --- android:googleLicensing:unspecified
+--- android:swipeList:unspecified
|    --- LOCAL: nineoldandroids-2.4.0.jar
+--- org.holoeverywhere:library:2.1.0
--- org.holoeverywhere:addon-preferences:2.1.0

release
+--- LOCAL: annotations.jar
+--- LOCAL: tagsoup-1.2.1.jar
+--- android:googleDownloading:unspecified
|    --- android:googleLicensing:unspecified
+--- android:swipeList:unspecified
|    --- LOCAL: nineoldandroids-2.4.0.jar
+--- org.holoeverywhere:library:2.1.0
--- org.holoeverywhere:addon-preferences:2.1.0
:dLight:androidDependencies
debug
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:dPro:androidDependencies
debug
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:googleDownloading:androidDependencies
debug
--- android:googleLicensing:unspecified

debugTest
+--- debug
|    --- android:googleLicensing:unspecified
--- android:googleLicensing:unspecified

release
--- android:googleLicensing:unspecified
:googleLicensing:androidDependencies
debug
No dependencies

debugTest
--- debug

release
No dependencies
:rPro:androidDependencies
debug
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0

debugTest
No dependencies

release
--- android:core:2.1.1-SNAPSHOT
      +--- LOCAL: tagsoup-1.2.1.jar
      +--- LOCAL: annotations.jar
      +--- android:googleDownloading:unspecified
      |    --- android:googleLicensing:unspecified
      +--- android:swipeList:unspecified
      |    --- LOCAL: nineoldandroids-2.4.0.jar
      +--- org.holoeverywhere:library:2.1.0
      --- org.holoeverywhere:addon-preferences:2.1.0
:swipeList:androidDependencies
debug
--- LOCAL: nineoldandroids-2.4.0.jar

debugTest
+--- LOCAL: nineoldandroids-2.4.0.jar
--- debug

release
--- LOCAL: nineoldandroids-2.4.0.jar

BUILD SUCCESSFUL

**Comment by Lakedaemon** [comment]
On 01/18/2014 01:23 PM, Sergey Shatunov wrote:

> First you should enable snapshot repository:
> 
> |holoeverywhere {
>    repository {
>      snapshot 'yes'
>    }
>    library {
>      version '2.1.1-SNAPSHOT'
>    }
> }
> |
> 
> Please describe here if it works.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/Prototik/HoloEverywhere/issues/759#issuecomment-32680812.
> 
> I made a little progress. By changing to

|holoeverywhere {
   repository {
     snapshot 'yes'
   }
   library {
     version = '2.1.1-SNAPSHOT'
   }
}|

It now, sets the dependency correctly (but doesn't find it). I now get 
this :

Failed to refresh Gradle project 'android' Could not find 
org.holoeverywhere:library:2.1.1-SNAPSHOT. Required by: 
android:core:unspecified

ubuntu@ubuntu-W150HRM:~/git/android$ ./gradlew androidDependencies
The TaskContainer.add() method has been deprecated and is scheduled to 
be removed in Gradle 2.0. Please use the create() method instead.

FAILURE: Build failed with an exception.
- What went wrong:
  A problem occurred configuring project ':core'.
  
  > Could not resolve all dependencies for configuration 
  > ':core:_debugCompile'.
  > Could not find org.holoeverywhere:library:2.1.1-SNAPSHOT.
  >   Required by:
  >       android:core:unspecified
  > Could not find org.holoeverywhere:addon-preferences:2.1.1-SNAPSHOT.
  >   Required by:
  >       android:core:unspecified
- Try:
  Run with --stacktrace option to get the stack trace. Run with --info or 
  --debug option to get more log output.

**Comment by Lakedaemon** [comment]
sadly, adding the following repositories doesn't help (it still won't 
find the libraries now that the dependencies are correctly set)

repositories {
     maven {
         url 'http://192.241.191.41/snapshot'
     }
}

or

repositories {
     mavenLocal()
}

(I checked out, built holoeverywhere and somehow exported it to the 
maven repository in /home/.m2/repository)

**Comment by Prototik** [comment]
Shit. I forget `=`.
I can success use a snapshot version for building with this config:

```
repositories {
     maven { url 'http://192.241.191.41/snapshot' }
}

holoeverywhere {
  repository {
    // Will fixed in 0.2.8 release
    // snapshot 'yes'
  }
  supportV4 {
    version = '19.0.3'
  }
  library {
    version = '2.1.1-SNAPSHOT'
  }
  addons {
    slider
    preferences
  }
}
```

My dependency tree looks like this:

```
:androidDependencies
debug
+--- org.holoeverywhere:library:2.1.1-SNAPSHOT
+--- org.holoeverywhere:addon-preferences:2.1.1-SNAPSHOT
\--- org.holoeverywhere:addon-slider:2.1.1-SNAPSHOT

debugTest
No dependencies

release
+--- org.holoeverywhere:library:2.1.1-SNAPSHOT
+--- org.holoeverywhere:addon-preferences:2.1.1-SNAPSHOT
\--- org.holoeverywhere:addon-slider:2.1.1-SNAPSHOT
```

**Comment by Lakedaemon** [comment]
I confirma that it works for me with the 0.2.8 plugin and this last 
configuration

|repositories {
      maven { url 'http://192.241.191.41/snapshot' }
}

holoeverywhere {
   repository {
     // Wil fixed in 0.2.9 release
     // snapshot 'yes'
   }
   supportV4 {
     version = '19.0.3'
   }
   library {
     version = '2.1.1-SNAPSHOT'
   }
   addons {
     slider
     preferences
   }
}|

Many thanks ! You are awesome !

**Comment by Prototik** [comment]
btw with 0.2.8 you can safely remove custom maven repo notation and even not use a `snapshot 'yes'` option - snapshot repo active by default.
